### PR TITLE
Add Py36, Drop Py33, restore tests for zope.component 4.4

### PR DESCRIPTION
**Pull request by jamadden**
Two commits:

- Coverage tox environment and coveralls (not 100% covered).
- Badges.
- Fix tests with zope.component 4.4.0.

- 100% coverage
- Whitespace and import cleanups
- Convert some doctests to real tests to assist with import cleanup and error messages.

### CODE REVIEW

File: src/zope/browserresource/metaconfigure.py [hunk 1]
```diff
@@ -60,14 +60,12 @@ def resource(_context, name, layer=IDefaultBrowserLayer,
 
     checker = NamesChecker(allowed_names, permission)
 
-    if (factory and (file or image or template)) or \
-       (file and (factory or image or template)) or \
-       (image and (factory or file or template)) or \
-       (template and (factory or file or image)):
+    too_many = bool(factory) + bool(file) + bool(image) + bool(template)
+    if too_many > 1:
```
Comment (icemac) [inline src/zope/browserresource/metaconfigure.py]: Wow!

File: src/zope/browserresource/tests/test_directives.py [hunk 2]
```diff
@@ -16,49 +16,57 @@
...
+
 from zope.publisher.interfaces.browser import IBrowserRequest
-from zope.publisher.interfaces.browser import IBrowserSkinType, IDefaultSkin
-from zope.security.proxy import removeSecurityProxy, ProxyFactory
-from zope.security.permission import Permission
-from zope.security.interfaces import IPermission
+
+from zope.security.proxy import ProxyFactory
+
+
```
Comment (icemac) [inline src/zope/browserresource/tests/test_directives.py]: Why are the so many blocks of imports separated by newlines? (sometimes one sometimes two)

File: src/zope/browserresource/tests/test_file.py [hunk 3]
```diff
@@ -18,18 +18,17 @@
 import os
 import re
 import unittest
+from email.utils import formatdate#, parsedate_tz, mktime_tz
```
Comment (icemac) [inline src/zope/browserresource/tests/test_file.py]: I think the code after the `#` can be removed.

File: src/zope/browserresource/tests/test_file.py [hunk 4]
```diff
@@ -72,228 +72,207 @@ def setUp(test):
...
-
-        >>> etag_maker = FileETag(object(), TestRequest())
-        >>> verifyObject(IETag, etag_maker)
-        True
-
-    By default we constuct an ETag from the file's mtime and size
 
-        >>> etag_maker(1234, 'abc')
-        '1234-3'
+    def setUp(self):
```
Comment (icemac) [inline src/zope/browserresource/tests/test_file.py]: I like it: converting a doctests into a unittests. 🥇

File: src/zope/browserresource/metaconfigure.py (hunk 1 above)
Comment (jamadden) [inline src/zope/browserresource/metaconfigure.py]: That made branch coverage look a lot better 😄

File: src/zope/browserresource/tests/test_directives.py (hunk 2 above)
Comment (jamadden) [inline src/zope/browserresource/tests/test_directives.py]: Hmm...I guess that when I clean up unused imports, I mostly just tend to delete the lines in place to try to make for a cleaner diff. If I need to move an import from "below" upwards, it tends to go at the end, unless there's already a clear grouping for its package, again in the effort to make the diff cleaner. Only if it starts to look real bad (subjective, I know) do I stop to organize the whole thing.

File: src/zope/browserresource/tests/test_file.py (hunk 3 above)
Comment (jamadden) [inline src/zope/browserresource/tests/test_file.py]: Whoops! Thanks.

**Comment by jamadden** [comment]
Thanks!

**Pull request merged**